pkg/osquery/interactive/interactive.go

```diff
@@ -0,0 +1,98 @@
+package interactive
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/kit/fs"
+	"github.com/kolide/launcher/pkg/osquery/table"
+	osquery "github.com/osquery/osquery-go"
+)
+
+const extensionName = "com.kolide.launcher_interactive"
+
+func StartProcess(rootDir, osquerydPath string, osqueryFlags []string) (*os.Process, error) {
+
+	if err := os.MkdirAll(rootDir, fs.DirMode); err != nil {
+		return nil, fmt.Errorf("creating root dir for interactive mode: %w", err)
+	}
+
+	// Get the current working directory.
+	cwd, err := os.Getwd()
+	if err != nil {
+		return nil, fmt.Errorf("error getting current working directory: %w", err)
+	}
+
+	// Transfer stdin, stdout, and stderr to the new process
+	// and also set target directory for the shell to start in.
+	pa := os.ProcAttr{
+		Files: []*os.File{os.Stdin, os.Stdout, os.Stderr},
+		Dir:   cwd,
+	}
+
+	socketPath := filepath.Join(rootDir, "osq.sock")
+
+	proc, err := os.StartProcess(osquerydPath, buildOsqueryFlags(socketPath, osqueryFlags), &pa)
+	if err != nil {
+		return nil, fmt.Errorf("error starting osqueryd in interactive mode: %w", err)
+	}
+
+	_, err = loadExtensions(socketPath, osquerydPath)
```

Comment:
Do we need to save the extension manager to stop it at the end?

Reply:
added call to shutdown extension manager server on exit